client/src/pages/Results/index.tsx

```diff
@@ -155,94 +92,78 @@ const ResultsPage = () => {
     }
   }, []);
 
-  const onResultClick = useCallback(
-    (repo?: string, path?: string) => {
-      navigate(
-        `/results?q=open:true ${
-          repo ? `repo:${encodeURIComponent(repo)}` : ''
-        } ${path ? `path:${encodeURIComponent(path)}` : ''}`,
-        { replace: !!openResult },
-      );
-      setIsFileNav(true);
-    },
-    [openResult],
-  );
+  const onResultClosed = useCallback(() => {
+    if (mode === FullResultModeEnum.SIDEBAR) {
+      setIsFiltersOpen(true);
+    }
+    setOpenResult(null);
+  }, [mode]);
 
-  const toggleFiltersOpen = useCallback(() => {
-    setIsFiltersOpen((prev) => !prev);
-  }, []);
+  useEffect(() => {
+    setSymbolsCollapsed(onlySymbolResults);
+  }, [onlySymbolResults]);
 
-  const handlePageChange = useCallback(
-    (page: number) => {
-      searchQuery(inputValue, page, globalRegex);
-      setPage(page);
-    },
-    [inputValue, globalRegex],
-  );
+  useEffect(() => {
+    if (page === 0) {
+      setTotalPages(resultsData.metadata.page_count);
+      setTotalCount(resultsData.metadata.total_count);
+    }
+    setFilters(mapFiltersData(resultsData.stats, filters));
+    setResults(mapResults(resultsData));
+    setPage(resultsData.metadata.page);
+  }, [resultsData]);
 
-  const renderPage = () => {
-    if (initialLoad) {
-      return <Skeleton />;
-    } else if (dirResult) {
-      return (
-        <RepositoryPage
-          sidebarOpen={!!openResult}
-          repository={{
-            name: dirResult.name,
-            fileCount: 0,
-            files: dirResult.entries,
-            commits: [],
-            url: '',
-            description: '',
-            branches: [],
-            followers: 1,
-            currentPath: dirResult.relativePath,
-          }}
-        />
-      );
+  useEffect(() => {
+    if (fileResultData) {
+      setOpenResult(mapFileResult(fileResultData.data[0]));
     }
-    return (
-      <>
-        <Filters isOpen={isFiltersOpen} toggleOpen={toggleFiltersOpen} />
-        <div
-          className="p-8 flex-1 overflow-x-auto mx-auto max-w-6.5xl box-content"
-          ref={ref}
-        >
-          <PageHeader
-            resultsNumber={totalCount || results.length}
-            showCollapseControls={onlySymbolResults}
-            loading={loading && !isFileNav}
-          />
-          {results.length || loading ? (
-            <ResultsList
-              results={results}
-              onResultClick={onResultClick}
-              page={page}
-              setPage={handlePageChange}
-              totalPages={totalPages}
-              loading={loading && !isFileNav}
-            />
-          ) : (
-            <NoResults suggestions={mockQuerySuggestions} />
-          )}
-        </div>
-      </>
-    );
-  };
+  }, [fileResultData]);
+
+  useEffect(() => {
+    // if (!isFileNav) {
+    //   ref.current?.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+    // }
+  }, [page]);
```

Comment:
this useEffect should either be not empty or removed

Reply:
fixed